File: Cpp/src/objects/output/DocOpinion.cpp
#include "DocOpinion.h"

DocOpinion::DocOpinion() {}
DocOpinion::~DocOpinion() {}

void DocOpinion::Serialize(Json::Value& root) {
    root["quotation"] = quotation;
    root["type"] = type;
    root["speaker"] = speaker;
    root["topic"] = topic;
    root["sentiment_score"] = sentiment_score;
    root["sentiment_polarity"] = sentiment_polarity;
}

void DocOpinion::Deserialize(Json::Value& root) {
    quotation = root.get("quotation", "").asString();
    type = root.get("type", "").asString();
    speaker = root.get("speaker", "").asString();
    topic = root.get("topic", "").asString();
    sentiment_score = root.get("sentiment_score", "").asDouble();
    sentiment_polarity = root.get("sentiment_polarity", "").asString();
}

File: dotNet/Semantria/Mapping/Output/DocOpinion.cs
using System;
using System.Collections.Generic;
using System.Runtime.Serialization;
using System.Xml.Serialization;
using Semantria.Com.Mapping.Output;

namespace Semantria.Com.Mapping.Output
{
	[DataContract(Name = "opinion")]
	[XmlRootAttribute("opinion", Namespace = "")]
	public class DocOpinion
	{
		[DataMember(Name = "quotation")]
		[XmlElementAttribute("quotation")]
		public string Label { get; set; }

		[DataMember(Name = "type")]
		[XmlElementAttribute("type")]
		public string Type { get; set; }

		[DataMember(Name = "speaker")]
		[XmlElementAttribute("speaker")]
		public string Speaker { get; set; }

		[DataMember(Name = "topic")]
		[XmlElementAttribute("topic")]
		public string Topic { get; set; }

		[DataMember(Name = "sentiment_score")]
		[XmlElementAttribute("sentiment_score")]
		public float SentimentScore { get; set; }

		[DataMember(Name = "sentiment_polarity")]
		[XmlElementAttribute("sentiment_polarity")]
		public string SentimentPolarity { get; set; }
	}
}
